src/Core/Foundation/PrestaShopVersion/PrestaShopVersion.php

```diff
@@ -0,0 +1,214 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\PrestaShop\Core\Foundation\PrestaShopVersion;
+
+use AppKernel;
+use PrestaShop\PrestaShop\Core\Foundation\PrestaShopVersion\Exception\InvalidVersionException;
+use InvalidArgumentException;
+
+class PrestaShopVersion
+{
+    const STRING = 0;
+    const INTEGER = 1;
+
+    /**
+     * Returns the current PrestaShop version which is hardcoded in \AppKernel::VERSION
+     *
+     * @return string For example "1.7.4.0"
+     */
+    public function getVersion()
+    {
+        return AppKernel::VERSION;
```

Comment:
Must be a parameter in constructor =)

Reply:
@PierreRambaud Thanks, I made a commit and it looks much better now :)